src/autodiff.jl

```diff
@@ -254,6 +138,9 @@ struct ControlJacobian{VI, VT, MT, SMT, VP, VD, SubT, SubD} <: AbstractJacobian
     end
 end
 
+struct StateJacobian <: AbstractJacobian end
```

Comment:
Could we move the definition of these two structures right after line 25 and the definition of `AbstractJacobian`? To ensure that we define first the types, then we proceed to the implementation.

Reply:
I would maybe go further, and use instead directly the `State` and `Control` type introduced in `src/model/model.jl`: 
https://github.com/exanauts/ExaPF.jl/blob/master/src/models/models.jl#L34-L50

Reply:
😬postponed.